### PR TITLE
Fix deprecated kernel options for newer distros

### LINKED ISSUE #83: [BUG] Do not use deprecated kernel boot arguments
**Description of problem:**
When installing Fedora-Rawhide (future Fedora 34) in Beaker, the following warnings are printed by the installer:

[    9.574010] dracut-cmdline[299]: Warning: 'ks=<URL>' 
[    9.584962] dracut-cmdline[299]: Warning: ks has been deprecated and will be removed. Please use inst.ks instead. 
[    9.596929] dracut-cmdline[299]: Warning: 'ksdevice=00:00:1A:1B:0B:8C' 
[    9.604937] dracut-cmdline[299]: Warning: ksdevice has been deprecated and will be removed. Please use inst.ks.device instead. 
[    9.637034] dracut-cmdline[299]: Warning: 'ksdevice=<MAC>' is deprecated. Supplied BOOTIF takes precedence. 
...
Boot argument 'leavebootorder' is deprecated and will be removed in the future. Please use 'inst.leavebootorder' instead.
...

Fedora change page:
https://fedoraproject.org/wiki/Changes/Ignore_Anaconda_kernel_boot_parameters_without_inst_prefix

Please, use the proper kernel command line arguments with the "inst." prefix instead of the deprecated arguments.

Boot arguments without the "inst." prefix are marked as deprecated since RHEL-7:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/installation_guide/chap-anaconda-boot-options#sect-boot-options-deprecated-removed

**Version-Release number of selected component (if applicable):**
Beaker 27.4

### PR DESCRIPTION
**Pull request by renanrodrigo**
This fixes #83.

Changed it for RHEL9+ and Fedora34+

### CODE REVIEW

**Comment by renanrodrigo** [comment]
Now to the correct target

**Comment by renanrodrigo** [comment]
@mdujava take a look

**Review by mdujava — APPROVED** [review]
LGTM

File: Server/bkr/server/model/scheduler.py [hunk 1]
```diff
@@ -2647,7 +2647,7 @@ def provision(self):
             if 'no_default_harness_repo' not in install_options.ks_meta \
                     and not self.harness_repo():
                 raise ValueError('Failed to find repo for harness')
-        ks_keyword = install_options.ks_meta.get('ks_keyword', 'ks')
+        ks_keyword = install_options.ks_meta.get('ks_keyword', 'inst.ks')
```
Comment (cbouchar) [inline Server/bkr/server/model/scheduler.py:2650]: shouldn't both 'ks' and 'inst.ks' be looked for?

File: Server/bkr/server/model/distrolibrary.py [hunk 2]
```diff
@@ -178,7 +178,17 @@ def default_install_options_for_distro(osmajor_name, osminor, variant, arch):
...
+                (fedora and fedora != 'rawhide' and int(fedora) < 33):
+            kernel_options['leavebootorder'] = None
+        else:
+            kernel_options['inst.leavebootorder'] = None
+
+    # Keep old kernel options for older distros
+    if rhel and int(rhel) < 8 or \
+            (fedora and fedora != 'rawhide' and int(fedora) < 33):
+        kernel_options['ksdevice'] = 'bootif'
+        ks_meta['ks_keyword'] = 'ks'
```
Comment (cbouchar) [inline Server/bkr/server/model/distrolibrary.py:191]: Should there be an else:
ks_meta['ks_keyword'] = 'inst.ks' ?????

**Review by cbouchar — CHANGES_REQUESTED** [review]
In the following files, don't references to 'ks' need to be given attention? 
Server/bkr/server/model/inventory.py
Server/bkr/server/systems.py

File: Server/bkr/server/model/scheduler.py (hunk 1 above)
Comment (StykMartin) [reply Server/bkr/server/model/scheduler.py:2650]: https://docs.python.org/3/library/stdtypes.html?highlight=dict#dict.get

**Comment by mdujava** [comment]
> In the following files, don't references to 'ks' need to be given attention?
> Server/bkr/server/model/inventory.py
> Server/bkr/server/systems.py

Good catch, those are reprovision on release and manual provision outside of job. 
https://github.com/beaker-project/beaker/blob/b755bf535df84a7efb3e30156ec557facae53f92/Server/bkr/server/model/inventory.py#L1549-L1555
https://github.com/beaker-project/beaker/blob/b755bf535df84a7efb3e30156ec557facae53f92/Server/bkr/server/systems.py#L228-L234

**Comment by renanrodrigo** [comment]
Looking into that, thanks

File: Server/bkr/server/model/distrolibrary.py (hunk 2 above)
Comment (renanrodrigo) [reply Server/bkr/server/model/distrolibrary.py:191]: yes, this is expected

**Comment by renanrodrigo** [comment]
@cbouchar @mdujava
Changed the ocurrences to look for ks_keyword as well.

File: IntegrationTests/src/bkr/inttest/server/test_model.py [hunk 3]
```diff
@@ -1400,6 +1400,26 @@ def test_ksmeta_ks_keyword(self):
...
+    def test_ks_option_for_older_systems(self):
+        distro_tree = data_setup.create_distro_tree(osmajor=u'CentOS7')
+        recipe = self.provision_distro_tree(distro_tree)
+        self.assertIn('ks=', recipe.installation.kernel_options)
+        self.assertNotIn('inst.ks=', recipe.installation.kernel_options)
+
+    def test_inst_ks_option_for_newer_systems(self):
+        distro_tree = data_setup.create_distro_tree(osmajor=u'Fedora34')
+        recipe = self.provision_distro_tree(distro_tree)
+        self.assertIn('inst.ks=', recipe.installation.kernel_options)
```
Comment (cbouchar) [inline IntegrationTests/src/bkr/inttest/server/test_model.py:1412]: Similar to older_systems test, should we test that 'ks=' is assertNotIn()???

**Review by cbouchar — CHANGES_REQUESTED** [review]
Other than my nit comment and too long a line issue, it looks good to me.

File: IntegrationTests/src/bkr/inttest/server/test_model.py (hunk 3 above)
Comment (renanrodrigo) [reply IntegrationTests/src/bkr/inttest/server/test_model.py:1412]: Well if 'inst.ks=' is there, 'ks=' will be there as well...

Comment (renanrodrigo) [reply IntegrationTests/src/bkr/inttest/server/test_model.py:1412]: asserted with a regex

**Review by cbouchar — APPROVED** [review]

**Pull request merged**